### DISCUSSION TITLE
Should I make 16k + records static or CSR?

### DISCUSSION THREAD

**Discussion by james-beston** [vercel/community#1149] [Help] [answered]
### Summary

I am creating a site that has pages for over 16k audiobooks. The information obviously never changes, so I was thinking these could all be static pages. But with the sheer number of pages that need creating and deploying, would I be better off serving them via API calls with SWR?

I know that I could cache the deploy, so I am not worried about a very long deployment time when it first goes to production. But I am not sure what would be best practice for so many records!

**Comment by james-beston** [comment, marked as answer]
Sorry for wasting your time! I found the answer (sort of) here https://github.com/vercel/next.js/discussions/21834

**Reply by amyegan** [reply]
Thanks for coming back to share your answer! ISR is a great solution
